src/main/java/org/breedinginsight/utilities/response/mappers/ProgramLocationQueryMapper.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.breedinginsight.utilities.response.mappers;
+
+import lombok.Getter;
+import org.breedinginsight.model.ProgramLocation;
+
+import javax.inject.Singleton;
+import java.util.Map;
+import java.util.function.Function;
+
+
+@Getter
+@Singleton
+public class ProgramLocationQueryMapper extends AbstractQueryMapper {
+
+    private Map<String, Function<ProgramLocation, ?>> fields;
+
+    public ProgramLocationQueryMapper() {
+        fields = Map.ofEntries(
+                Map.entry("name", ProgramLocation::getName),
+                Map.entry("abbreviation", ProgramLocation::getAbbreviation),
+                Map.entry("slope", ProgramLocation::getSlope),
+                Map.entry("createdAt", ProgramLocation::getCreatedAt),
+                Map.entry("updatedAt", ProgramLocation::getUpdatedAt),
+                Map.entry("createdByUserId",
+                        (location) -> location.getCreatedByUser() != null ? location.getCreatedByUser().getId() : null),
+                Map.entry("createdByUserName",
+                        (location) -> location.getCreatedByUser() != null ? location.getCreatedByUser().getName() : null),
+                Map.entry("updatedByUserId",
+                        (location) -> location.getUpdatedByUser() != null ? location.getUpdatedByUser().getId() : null),
+                Map.entry("updatedByUserName",
```

Comment:
I probably should have put this comment on #44 as well, but, it may be worth changing these strings to constants in this class, or maybe even a nested enum.  Same thing goes for the other `QueryMapper`s

Reply:
I was thinking since these are only used in this one place it wouldn't add any benefit defining those strings as constants. Since the incoming request strings get mapped to the getters, the map keys aren't referenced any where in the code. So normally I agree with the constants are enums pattern, but I was thinking it wasn't necessary here. What are your thoughts on it?

Reply:
Ok, good point.  Something to keep in mind if there's ever a direct use of the mapped fields in the future.